better-politics-mod/events/bpm_east_asia_dynamics.txt

```diff
@@ -0,0 +1,209 @@
+# Great Qing loses the opium war and all related countries will get this event
+namespace = bpm_east_asia_dynamics
+bpm_east_asia_dynamics.1 = {
+	type = country_event
+	placement = ROOT
+	title = bpm_east_asia_dynamics.1.t
+	desc = {
+		first_valid = {
+			triggered_desc = {
+				desc = bpm_east_asia_dynamics.1.d
+				trigger = {
+					scope:china_scope = {
+						has_law = law_type:law_free_trade
+					}
+				}
+			}
+			triggered_desc = {
+				desc = bpm_east_asia_dynamics.1.d2
+				trigger = {
+					exists = scope:treaty_port_state
+				}
+			}
+			triggered_desc = {
+				desc = bpm_east_asia_dynamics.1.d3
+				trigger = {
+					always = yes
+				}
+			}
+		}
+	}
+	flavor = bpm_east_asia_dynamics.1.f
+	duration = 3
+	event_image = {
+		video = "asia_westerners_arriving"
+	}
+	on_created_soundeffect = "event:/SFX/UI/Alerts/event_appear"
+	icon = "gfx/interface/icons/event_icons/event_newspaper.dds"
+	trigger = {
+		# triggered by JE complete: zz_bpm_00_opium_wars.txt
+	}
+	immediate = {
+		c:CHI = {
+			save_scope_as = china_scope
+		}
+
+		random_country = {
+			limit = {
+				has_variable = opium_wars_aggressor
+			}
+			save_scope_as = opium_wars_aggressor
+		}
+
+		scope:opium_wars_aggressor = {
+			limit = {
+				any_scope_state = {
+					market = {
+						owner = scope:china_scope
+					}
+				}
+			}
+			save_scope_as = opium_wars_treaty_port_winner
+			random_scope_state = {
+				limit = {
+					is_treaty_port = yes
+					state_region = {
+						any_scope_state = {
+							owner = scope:china_scope
+						}
+					}
+				}
+				save_scope_as = treaty_port_state
+			}
+		}
+	}
+	# We shall stand with the Middle Kingdom!
+	option = {
+		name = bpm_east_asia_dynamics.1.a
+		trigger = {
+			OR = {
+				is_subject_of = scope:china_scope
+				AND = {
+					NOT = {
+						has_law = law_type:law_isolationism
+					}
+					scope:china_scope = {
+						relations:root >= relations_threshold:neutral
+					}
+				}
+			}
+		}
+		add_radicals = {
+			value = large_radicals
+		}
+		# add_modifier = {
+		#     name = bpm_east_asia_dynamics_opium_wars_stability
+		#     months = normal_modifier_time
+		#     is_decaying = yes
+		# }
```

Comment:
Commented because it seems that the `country_rigidity_baseline_add` effect does not exist atm in this branch